updates on mom6 timesteps (#1504)
COSIMA/MOM6-CICE6: documentation/docs/pages/infrastructure/NUOPC-driver.md

### MOM6 time-steps

> MOM6 has 5 timesteps - see [here](https://youtu.be/JKMwd8VXYcU?t=383) and [here](https://youtu.be/JKMwd8VXYcU?t=2165) and [here](https://mom6.readthedocs.io/en/main/api/generated/pages/Timestep_Overview.html). From shortest to longest they are: barotropic, baroclinic (Lagrangian), thermodynamic, tracer advection and vertical remapping. Of these, it is common to set at least these 3 timesteps in the `MOM_input` file:

Reviewer: It's probably worth mentioning that the remapping timestep cannot currently be set independently of the thermo timestep.

```suggestion
MOM6 has 5 timesteps - see [here](https://youtu.be/JKMwd8VXYcU?t=383) and [here](https://youtu.be/JKMwd8VXYcU?t=2165) and [here](https://mom6.readthedocs.io/en/main/api/generated/pages/Timestep_Overview.html). From shortest to longest they are: barotropic, baroclinic (Lagrangian), thermodynamic, tracer advection and vertical remapping (note that currently in MOM6 remapping is performed as part of the thermodynamic step so their timesteps cannot be set independently). It is common to set at least these 3 timesteps in the `MOM_input` file:
```

> MOM6 has 5 timesteps - see [here](https://youtu.be/JKMwd8VXYcU?t=383) and [here](https://youtu.be/JKMwd8VXYcU?t=2165) and [here](https://mom6.readthedocs.io/en/main/api/generated/pages/Timestep_Overview.html). From shortest to longest they are: barotropic, baroclinic (Lagrangian), thermodynamic, tracer advection and vertical remapping (note that currently in MOM6 remapping is performed as part of the thermodynamic step so their timesteps cannot be set independently). It is common to set at least these 3 timesteps in the `MOM_input` file:

Reviewer: I think if you were going to set the tracer advection timestep, you'd typically set it to less than `DT_THERM`

```suggestion
MOM6 has 5 timesteps - see [here](https://youtu.be/JKMwd8VXYcU?t=383) and [here](https://youtu.be/JKMwd8VXYcU?t=2165) and [here](https://mom6.readthedocs.io/en/main/api/generated/pages/Timestep_Overview.html). From shortest to longest they are: barotropic, baroclinic (Lagrangian), tracer advection, thermodynamic and vertical remapping (note that currently in MOM6 remapping is performed as part of the thermodynamic step so their timesteps cannot be set independently). It is common to set at least these 3 timesteps in the `MOM_input` file:
```

> MOM6 has 5 timesteps - see [here](https://youtu.be/JKMwd8VXYcU?t=383) and [here](https://youtu.be/JKMwd8VXYcU?t=2165) and [here](https://mom6.readthedocs.io/en/main/api/generated/pages/Timestep_Overview.html). From shortest to longest they are: barotropic, baroclinic (Lagrangian), tracer advection, thermodynamic and vertical remapping (note that currently in MOM6 remapping is performed as part of the thermodynamic step so their timesteps cannot be set independently). It is common to set at least these 3 timesteps in the `MOM_input` file:

Reviewer 2: Not sure if it's helpful to describe it as "5" when thermodynamic and vertical remapping can't be set independently.

Author: The thinking here is that the docs and code refer to them seperately I think ? even though they can't currently be different 

(is that right @dougiesquire ? )

Reviewer: Not the code. But docs and videos do all over the place. So I think it's good to list as a separate timestep to be consistent with those, but note that it currently is the same as `DT_THERM` (as is already done)

Reviewer 2: Sounds good.

> MOM6 has 5 timesteps - see [here](https://youtu.be/JKMwd8VXYcU?t=383) and [here](https://youtu.be/JKMwd8VXYcU?t=2165) and [here](https://mom6.readthedocs.io/en/main/api/generated/pages/Timestep_Overview.html). From shortest to longest they are: barotropic, baroclinic (Lagrangian), tracer advection, thermodynamic and vertical remapping (note that currently in MOM6 remapping is performed as part of the thermodynamic step so their timesteps cannot be set independently). It is common to set at least these 3 timesteps in the `MOM_input` file:

Reviewer 2: ```suggestion
MOM6 has 5 timesteps. Useful references include:

  - [here](https://youtu.be/JKMwd8VXYcU?t=383); 
  - [here](https://youtu.be/JKMwd8VXYcU?t=2165);
  - [here](https://mom6.readthedocs.io/en/main/api/generated/pages/Timestep_Overview.html). 
  
  From shortest to longest they are: barotropic, baroclinic (Lagrangian), tracer advection, thermodynamic and vertical remapping (note that currently in MOM6 remapping is performed as part of the thermodynamic step so their timesteps cannot be set independently). It is common to set at least these 3 timesteps in the `MOM_input` file:
```

Or something like this. Previous version had two "and"s and the wrong kind of dash.

Reviewer 2: Then for this bit:
> From shortest to longest they are: barotropic, baroclinic (Lagrangian), tracer advection, thermodynamic and vertical remapping (note that currently in MOM6 remapping is performed as part of the thermodynamic step so their timesteps cannot be set independently).

I'd suggest it's worth saying in this list where they all are. Something like:

> From shortest to longest they are: barotropic (`DTBT`), baroclinic Lagrangian (`DTBT`), tracer advection (`DT_TRACER_ADVECT`), thermodynamic (`DT_THERM`) and vertical remapping. Note that currently in MOM6 remapping is performed as part of the thermodynamic step so their timesteps cannot be set independently.

On tracer advect: https://github.com/ACCESS-NRI/access-om3-configs/blob/8f26a86d1b1c076f5abbd1f975ebae953d8660d7/docs/MOM_parameter_doc.all#L85

Is there somewhere one can set the thermodynamic/remapping timestep?

Reviewer: `DT_THERM` is the thermodynamic time step

Reviewer: It say this a few lines further down

Reviewer 2: Oh right of course. I've updated [my proposed text](https://github.com/ACCESS-NRI/access-om3-configs/pull/1504#discussion_r3575959453)

Author: I implemented a version of this now

Reviewer 2: I think you mean this [3342133](https://github.com/ACCESS-NRI/access-om3-configs/pull/1504/commits/334213382afc433cca642ad4a2fa96da6d5dda4f)?

That's an improvement but can you consider [this](https://github.com/ACCESS-NRI/access-om3-configs/pull/1504#discussion_r3575959453) suggestion too, i.e.
> From shortest to longest they are: barotropic (DTBT), baroclinic Lagrangian (DTBT), tracer advection (DT_TRACER_ADVECT), thermodynamic (DT_THERM) and vertical remapping. Note that currently in MOM6 remapping is performed as part of the thermodynamic step so their timesteps cannot be set independently.

I think naming the timesteps explicitly is helpful from the outset.

Reviewer 2: Resolved:
 - [6d7c83e](https://github.com/ACCESS-NRI/access-om3-configs/pull/1504/commits/6d7c83ef2a3348ef9640c853839ce0e40c70f0b8)
 - [0c9e031](https://github.com/ACCESS-NRI/access-om3-configs/pull/1504/commits/0c9e03118105a40f4e73e903eca8647132190578)

- Barotropic time-step (`DTBT`) for integration of sea surface and depth-averaged horizontal velocity. If set negative (e.g. `DTBT = -0.95`), the magnitude of `DTBT` is interpreted a fraction of the stability limit, so can be set independently of the model configuration (e.g. resolution). `DTBT_RESET_PERIOD` controls how often the stability limit is recalculated.

- Baroclinic time-step (`DT`) for Lagrangian stacked shallow-water equations; often called "the" model timestep; needs to be short enough to resolve internal gravity waves, inertial oscillations and advection on the horizontal grid (i.e. this is resolution-dependent).

> - Thermodynamics time-step (`DT_THERM`), which can be [set to resolve the relevant physics](https://youtu.be/JKMwd8VXYcU?si=dBk09lMVPsMF3aJT&t=541) (e.g. one to three hours or so to capture the diurnal cycle), independent of the horizontal grid resolution. It is possible to set `DT_THERM` longer than the coupling time-step, but with [`DIABATIC_FIRST = False`](https://github.com/search?q=repo%3AACCESS-NRI%2Faccess-om3-configs+path%3Adoc%2FMOM_input+content%3ADIABATIC_FIRST&type=code).

Reviewer: ```suggestion
- Thermodynamics time-step (`DT_THERM`), which can be [set to resolve the relevant physics](https://youtu.be/JKMwd8VXYcU?si=dBk09lMVPsMF3aJT&t=541) (e.g. one to three hours or so to capture the diurnal cycle), independent of the horizontal grid resolution. It is possible to set `DT_THERM` longer than the coupling time-step, but only with [`DIABATIC_FIRST = False`](https://github.com/search?q=repo%3AACCESS-NRI%2Faccess-om3-configs+path%3Adoc%2FMOM_input+content%3ADIABATIC_FIRST&type=code).
```

> - Thermodynamics time-step (`DT_THERM`), which can be [set to resolve the relevant physics](https://youtu.be/JKMwd8VXYcU?si=dBk09lMVPsMF3aJT&t=541) (e.g. one to three hours or so to capture the diurnal cycle), independent of the horizontal grid resolution. It is possible to set `DT_THERM` longer than the coupling time-step, but only with [`DIABATIC_FIRST = False`](https://github.com/search?q=repo%3AACCESS-NRI%2Faccess-om3-configs+path%3Adoc%2FMOM_input+content%3ADIABATIC_FIRST&type=code).

Reviewer: ```suggestion
- Thermodynamics time-step (`DT_THERM`), which can be [set to resolve the relevant physics](https://youtu.be/JKMwd8VXYcU?si=dBk09lMVPsMF3aJT&t=541) (e.g. one to three hours or so to capture the diurnal cycle), independent of the horizontal grid resolution. It is possible to set `DT_THERM` longer than the coupling time-step, but only with [`DIABATIC_FIRST = False`](https://github.com/search?q=repo%3AACCESS-NRI%2Faccess-om3-configs+path%3Adoc%2FMOM_input+content%3ADIABATIC_FIRST&type=code). The tracer advection timestep (`DT_TRACER_ADVECT`) defaults to the same as `DT_THERM`.
```

> - Thermodynamics time-step (`DT_THERM`), which can be [set to resolve the relevant physics](https://youtu.be/JKMwd8VXYcU?si=dBk09lMVPsMF3aJT&t=541) (e.g. one to three hours or so to capture the diurnal cycle), independent of the horizontal grid resolution. It is possible to set `DT_THERM` longer than the coupling time-step, but only with [`DIABATIC_FIRST = False`](https://github.com/search?q=repo%3AACCESS-NRI%2Faccess-om3-configs+path%3Adoc%2FMOM_input+content%3ADIABATIC_FIRST&type=code). The tracer advection timestep (`DT_TRACER_ADVECT`) defaults to the same as `DT_THERM`.

Reviewer 2: "Thermodynamics time-step (`DT_THERM`), which can be [set to resolve the relevant physics](https://youtu.be/JKMwd8VXYcU?si=dBk09lMVPsMF3aJT&t=541) ... independent of the horizontal grid resolution. "

This reads to me like one might want to set the same `DT_THERM` across resolutions? In practice it's shorter (??) at the higher resolutions ..

| Branch | Link to `MOM_input` | `DT_THERM` (s) |
|---|---|---|
| `dev-MC_100km_jra_ryf` | https://github.com/ACCESS-NRI/access-om3-configs/blob/dev-MC_100km_jra_ryf/MOM_input | 3600.0 |
| `dev-MC_100km_jra_iaf` | https://github.com/ACCESS-NRI/access-om3-configs/blob/dev-MC_100km_jra_iaf/MOM_input | 3600.0 |
| `dev-MC_25km_jra_ryf` | https://github.com/ACCESS-NRI/access-om3-configs/blob/dev-MC_25km_jra_ryf/MOM_input | 7200.0 |
| `dev-MC_25km_jra_iaf` | https://github.com/ACCESS-NRI/access-om3-configs/blob/dev-MC_25km_jra_iaf/MOM_input | 7200.0 |
| `release-MC_25km_jra_ryf` | https://github.com/ACCESS-NRI/access-om3-configs/blob/release-MC_25km_jra_ryf/MOM_input | 7200.0 |
| `dev-MC_8km_jra_ryf` | https://github.com/ACCESS-NRI/access-om3-configs/blob/dev-MC_8km_jra_ryf/MOM_input | 7200.0 |

Author: Yes - my understanding is that it should/could be resolution independent

Reviewer 2: I think @dougiesquire might have made this comment on Monday but could be something for @ezhilsabareesh8 to try with the current optimisation work?

### MOM6 time-steps

MOM6 has 5 timesteps - see these references:

- [Lecture: MOM6 time stepping (Vertical Langrangian Remap)](https://youtu.be/JKMwd8VXYcU?t=383)
- [Lecture: MOM6 time stepping (Summary)](https://youtu.be/JKMwd8VXYcU?t=2165)
- [MOM6 ReadtheDocs](https://mom6.readthedocs.io/en/main/api/generated/pages/Timestep_Overview.html). 

> From shortest to longest they are: barotropic, baroclinic (Lagrangian), tracer advection, thermodynamic and vertical remapping (note that currently in MOM6 remapping is performed as part of the thermodynamic step so their timesteps cannot be set independently). It is common to set at least these 3 timesteps in the `MOM_input` file:

Author: ```suggestion
From shortest to longest they are: barotropic (DTBT), baroclinic Lagrangian (DTBT), tracer advection (DT_TRACER_ADVECT), thermodynamic (DT_THERM) and vertical remapping. 
Note that currently in MOM6 remapping is performed as part of the thermodynamic step so their timesteps cannot be set independently. 
It is common to set at least these 3 timesteps in the `MOM_input` file:
```

### MOM6 time-steps

MOM6 has 5 timesteps - see these references:

- [Lecture: MOM6 time stepping (Vertical Langrangian Remap)](https://youtu.be/JKMwd8VXYcU?t=383)
- [Lecture: MOM6 time stepping (Summary)](https://youtu.be/JKMwd8VXYcU?t=2165)
- [MOM6 ReadtheDocs](https://mom6.readthedocs.io/en/main/api/generated/pages/Timestep_Overview.html). 

> From shortest to longest they are: barotropic (DTBT), baroclinic Lagrangian (DTBT), tracer advection (DT_TRACER_ADVECT), thermodynamic (DT_THERM) and vertical remapping. 

Author: ```suggestion
From shortest to longest they are: barotropic (`DTBT`), baroclinic Lagrangian (`DTBT`), tracer advection (`DT_TRACER_ADVECT`), thermodynamic (`DT_THERM`) and vertical remapping. 
```

> From shortest to longest they are: barotropic (`DTBT`), baroclinic Lagrangian (`DTBT`), tracer advection (`DT_TRACER_ADVECT`), thermodynamic (`DT_THERM`) and vertical remapping. 

Reviewer: ```suggestion
From shortest to longest they are: barotropic (`DTBT`), baroclinic Lagrangian (`DT`), tracer advection (`DT_TRACER_ADVECT`), thermodynamic (`DT_THERM`) and vertical remapping. 
```

Reviewer 2: good catch!
